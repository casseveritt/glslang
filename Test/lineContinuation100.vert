#version 100

// non-line continuation comment \
#error good error



float f\
oo;  // same as 'float foo;'

#error e2

#define MAIN void main() \
   {                     \
gl_Position = vec4(foo); \
} 

#error e3

MAIN

vec4 foo2(vec4 a)
{                                
  vec4 b = a;       \
  return b;                   
}

// aoeuntheo unatehutna \ antaehnathe 
// anteonuth $ natohe " '
// anteonuth     natohe
/*@*/
/* *@/*/
//@

#define A int q1 = \ 1
#define B int q2 = \1
#define C int q3 = $ 1
#define D int q4 = @ 1

int a1 = \ 4;  // ERROR
int a2 = @ 3;  // ERROR
int a3 = $4;   // ERROR
int a4 = a2\;  // ERROR

A;
B;
C;
D;

# \

# \
    error bad continuation

#define QUOTE "ab\
cd"
