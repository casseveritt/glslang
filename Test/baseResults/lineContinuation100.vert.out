lineContinuation100.vert
WARNING: 0:3: 'line continuation' : used at end of comment, but this version does not provide line continuation 
ERROR: 0:4: '#error' : good error  
ERROR: 0:8: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:11: '#error' : e2  
ERROR: 0:13: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:14: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:15: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:18: '#error' : e3  
ERROR: 0:24: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:40: '\' : illegal use of escape character 
ERROR: 0:41: '@' : unexpected token 
ERROR: 0:42: '$' : unexpected token 
ERROR: 0:43: '\' : illegal use of escape character 
ERROR: 0:45: '\' : illegal use of escape character 
ERROR: 0:46: '\' : illegal use of escape character 
ERROR: 0:47: '$' : unexpected token 
ERROR: 0:48: '@' : unexpected token 
ERROR: 0:50: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:52: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 0:53: '#error' : bad continuation  
ERROR: 0:55: 'line continuation' : not supported for this version or the enabled extensions 
ERROR: 20 compilation errors.  No code generated.


ERROR: node is still EOpNull!
0:20  Function Definition: main( (void)
0:20    Function Parameters: 
0:20    Sequence
0:20      move second child to first child (highp 4-component vector of float)
0:20        'gl_Position' (gl_Position highp 4-component vector of float)
0:20        Construct vec4 (highp 4-component vector of float)
0:20          'foo' (highp float)
0:22  Function Definition: foo2(vf4; (highp 4-component vector of float)
0:22    Function Parameters: 
0:22      'a' (in highp 4-component vector of float)
0:24    Sequence
0:24      Sequence
0:24        move second child to first child (highp 4-component vector of float)
0:24          'b' (highp 4-component vector of float)
0:24          'a' (in highp 4-component vector of float)
0:25      Branch: Return with expression
0:25        'b' (highp 4-component vector of float)
0:40  Sequence
0:40    move second child to first child (highp int)
0:40      'a1' (highp int)
0:40      Constant:
0:40        4 (const int)
0:41  Sequence
0:41    move second child to first child (highp int)
0:41      'a2' (highp int)
0:41      Constant:
0:41        3 (const int)
0:42  Sequence
0:42    move second child to first child (highp int)
0:42      'a3' (highp int)
0:42      Constant:
0:42        4 (const int)
0:43  Sequence
0:43    move second child to first child (highp int)
0:43      'a4' (highp int)
0:43      'a2' (highp int)
0:45  Sequence
0:45    move second child to first child (highp int)
0:45      'q1' (highp int)
0:45      Constant:
0:45        1 (const int)
0:46  Sequence
0:46    move second child to first child (highp int)
0:46      'q2' (highp int)
0:46      Constant:
0:46        1 (const int)
0:47  Sequence
0:47    move second child to first child (highp int)
0:47      'q3' (highp int)
0:47      Constant:
0:47        1 (const int)
0:48  Sequence
0:48    move second child to first child (highp int)
0:48      'q4' (highp int)
0:48      Constant:
0:48        1 (const int)
0:?   Linker Objects
0:?     'foo' (highp float)
0:?     'a1' (highp int)
0:?     'a2' (highp int)
0:?     'a3' (highp int)
0:?     'a4' (highp int)
0:?     'q1' (highp int)
0:?     'q2' (highp int)
0:?     'q3' (highp int)
0:?     'q4' (highp int)


Linked vertex stage:



